src/esp/sim/Simulator.cpp

```diff
@@ -638,10 +640,37 @@ bool Simulator::recomputeNavMesh(nav::PathFinder& pathfinder,
     return false;
   }
 
+  if (&pathfinder == pathfinder_.get()) {
+    if (navMeshVisNode_ != nullptr) {
+      // if updating pathfinder_ instance, reset the visualization if necessary.
+      toggleNavMeshVisualization();  // clear old version
+      toggleNavMeshVisualization();
+    }
+  }
+
   LOG(INFO) << "reconstruct navmesh successful";
   return true;
 }
 
+void Simulator::toggleNavMeshVisualization() {
+  if (pathfinder_ == nullptr)
+    return;
+  if (navMeshVisNode_ == nullptr && pathfinder_->isLoaded()) {
+    auto& sceneGraph = sceneManager_->getSceneGraph(activeSceneID_);
+    auto& rootNode = sceneGraph.getRootNode();
+    auto& drawables = sceneGraph.getDrawables();
+    navMeshVisNode_ = &rootNode.createChild();
+    navMeshVisPrimID_ = resourceManager_->loadNavMeshVisualization(
```

Comment:
I see that loadNavMeshVisualization can return ID_UNDEFINED. I guess removePrimitiveMesh will fail a CHECK on the next toggle. Does that sound like the behavior we want?

Reply:
Good point, thanks. 👍